### DISCUSSION TITLE
Restoring backups from different environments

### DISCUSSION THREAD

**Discussion by cllovatto** [longhorn/longhorn#3393] [Q&A] [answered]
I want to restore a backup from production into staging. Problem is that Longhorn volumes names are different between those two environments. I tried to edit the restored volume name, but it failed.

Is there a way to achieve that?

**Comment by jenting** [comment, marked as answer]
Why do you what the restored Longhorn volumes name to be the same?
As long as the PVC/PV are the same, the application could use the original backed data.

**Reply by cllovatto** [reply]
That was what I ended up doing. For now, that solution is ok for me. 

For context, Staging environment is a new one, I refactored many things in my IaC and wanted a better naming for my volumes. Having production data in Staging environment is a requirement. Problem is, I could not find a way to change the volume name without losing data. Could not change existing volume name in cluster and if I create one with a new name, cannot restore any of the backups. I am stuck with the name I chose before.

**Reply by jenting** [reply]
Okay, I don't know the historical reason that why Longhorn does not have a way to change the volume name _or_ restore backup with the original volume name.

@shuo-wu @joshimoo do you know the context?

**Reply by shuo-wu** [reply]
For **existing** volumes, there are corresponding CRs in k8s hence the names are immutable no matter if the volumes are restored from backups or newly created ones. But Longhorn allows using a different name **when** restoring a volume from backups.
